models/src/main/java/com/vimeo/networking2/Membership.kt

```diff
@@ -0,0 +1,43 @@
+package com.vimeo.networking2
+
+import com.squareup.moshi.Json
+import com.squareup.moshi.JsonClass
+import com.vimeo.networking2.enums.AccountType
+import com.vimeo.networking2.enums.asEnum
+
+/**
+ * This object contains user membership information.
+ */
+@JsonClass(generateAdapter = true)
+data class Membership(
+
+    /**
+     * The user's membership level
+     */
+    @Json(name = "display")
+    val display: String,
+
+    /**
+     * The user's account type.
+     */
+    @Json(name = "type")
+    val type: String,
+
+    /**
+     * Information about the user's badge.
+     */
+    @Json(name = "badge")
+    val badge: UserBadge? = null,
+
+    /**
+     * Information about the user's subscription.
+     */
+    @Json(name = "subscription")
+    val subscription: Subscription? = null
+)
+
+/**
+ * @see AccountType
+ */
+val Membership.accountType: AccountType
```

Comment:
Glad to see the extension pattern being put to use. In order to follow the "type" pattern we've used elsewhere, a few minor changes need to be made:
- `type` should be renamed to `rawType`
- `accountType` should be renamed to `type`
- `AccountType` should be renamed to `MembershipType`
- this extension function should also have a `@see` pointing back to the raw string property
- the raw string property should have an `@see pointing to the extension property

Reply:
✔️ I did 1,2,4,5
Regarding:
> AccountType should be renamed to MembershipType

I think "account type" is used most places instead of "membership type", so even though this is within a `membership.type` field I think it makes sense to stick with the naming used by the API, in our docs, etc.

Even the official API docs for the field is:
> description: "The user's account type:... "

We could also choose to keep the extension property as `accountType` based on this. Let me know what you think.